internal/build/docker_test.go

```diff
@@ -492,6 +492,49 @@ func TestBuildDockerWithStepsFromExistingPreservesEntrypoint(t *testing.T) {
 	f.assertFilesInContainer(f.ctx, cID, expected)
 }
 
+func TestBuildDockerWithStepsFromExistingRemovesPreviousMounts(t *testing.T) {
```

Comment:
seems like lots of this test case is irrelevant to the thing your testing for. don't need steps or entrypoint, and comment at ln. 512 is now obsolete. also since we're not testing entrypoint you can just use `assertFilesInImage`